tests/table/test_upsert.py

```diff
@@ -478,8 +485,8 @@ def test_upsert_with_duplicate_rows_in_table(catalog: Catalog) -> None:
         schema=arrow_schema,
     )
 
-    with pytest.raises(ValueError, match="Target table has duplicate rows, aborting upsert"):
-        _ = tbl.upsert(df)
+    # with pytest.raises(ValueError, match="Target table has duplicate rows, aborting upsert"):
+    #     _ = tbl.upsert(df)
```

Comment:
We should be able to let this test pass pretty easily: https://github.com/apache/iceberg-python/blob/b85127e05b699f5e8f7acc1034b9f258fd209477/pyiceberg/table/upsert_util.py#L67